### ISSUE TITLE
[REQUEST CODE] For Coolmoon ARGB LED/FAN controllers remote

### ISSUE THREAD

**Issue by heit0r** [irplus-remote/irplus-codes.github.io#643]
Coolmoon is a chinese brand of ARGB LED/FAN controllers. They also use the same LTG1596 board like many other brands. A friend said they copied a Corsair ARGB controller. Anyways, my ir remote don't work anymore. If anyone have one and could share the code, I'll be very glad.

**Comment by brunohenriquy** [comment]
> Coolmoon is a chinese brand of ARGB LED/FAN controllers. They also use the same LTG1596 board like many other brands. A friend said they copied a Corsair ARGB controller. Anyways, my ir remote doesn't work anymore. If anyone have one and could share the code, I'll be very glad.

Any luck with this? I have a Coolmoon controller and the remote, I am trying to clone it with a Tuya IR smart module but no success.

**Comment by heit0r** [comment]
Not at all. After opening the remote to see if I could fix anything, I stumble across an unlabeled chip (IC1) and an unusual circuit for an IR remote:

![01](https://github.com/user-attachments/assets/3519c653-3e33-40c9-b058-4951d9a2cd86)

The LED controller itself has also another unlabeled chip (U3), but U1 caught my attention:

![03](https://github.com/user-attachments/assets/2216f9d4-62ec-494e-94eb-554aec902756)
![04](https://github.com/user-attachments/assets/66903cce-eb70-4250-9c47-d062405631c2)

According to its datasheet (from chinese by google translator) this is a RF (radio frequency) receiver. 433 MHz is the same band of garage doors, parking lot entrances, car keys and lots of other devices. This is an internationally free band.

![06](https://github.com/user-attachments/assets/4ccadf49-e1db-4161-9616-e5fec8694482)

After a couple of pages, here's the circuit where you can see the antenna path (from pin 2 to ANT1) and its filtering with capacitors and inductors. You can also confirm the crystal value (Y1) is the same in the datasheet for the 433MHz configuration:

![07](https://github.com/user-attachments/assets/39d39c10-1b9b-4859-bb65-9995bf2dfa0b)

This explains the lack of copper plate (transparency) on the board, so the antenna can pickup the radio frequency emmited by the supposed IR remote.

![08](https://github.com/user-attachments/assets/d49e004e-55f8-4b63-9e02-1cc1dcdaee8c)

What should be the antenna path, most likely:
![09](https://github.com/user-attachments/assets/3dd7509b-9117-4630-a441-c41a76c9f033)

The manufacturer maybe went on RF probably because there's no way for an IR beam to strongly pass through glass and wires that can easily block and reflect it in its way to the IR sensor. RF can pass through almost anything without much problem. But RF is totally beyond my reach in knowledge and equipment.

BUT there might be a solution that I didn't try yet:

According with the ARGB standard, here you have 3 pads: 5V+ (the square one), data+ and GND. But the 5V is disconnected, because the LED controller is already being fed by the computer's PSU. So, in practice, you only have d+ and the ground. Be sure which one is GND by testing it with a multimeter.

![10](https://github.com/user-attachments/assets/64c90d5c-5861-46eb-8449-85fff956b865)

You can interface it with an, for example, Arduino Nano with this project: https://github.com/pavitra14/ArduinoARGBSync
I have an Arduino Nano but not the time to make it all happen. If you are willing to do so and get any luck on this, please let me know. When I try this, I'll post the outcome here.

The only thing that I could do so far is to disconnect the reset switch from the motherboard and connect it to the S1 switch (at the left side of the SATA-POWER connector) in the led controller. This is easy to do and there's no reverse polarity, so don't worry. By doing this you can, at least, change between the (ugly) programmed colors in the controller by pressing the reset button on your case.

BTW, the manufacturer should mention this is a RF remote. The LED just blinks a little bit to give the impression this is a real IR remote, which isn't true at all. Total scam 🤡

Sorry the long post. But maybe others can join the discussion and go on from where I stopped. Or not. :)

**Comment by brunohenriquy** [comment]
> BTW, the manufacturer should mention this is a RF controller. The LED just blinks a little bit to give the impression this is a real IR controller, which isn't true at all. Total scam 🤡
> 
> Sorry the long post. But maybe others can join the discussion and go on from where I stopped. Or not. :)

Wow my friend!!! Just wow....

You've done an awesome reverse engineering job.

"Sorry the long post"???? Come on, I loved it!!

Thank you so much for this. I will definitely try to interface it with an Arduino Nano, or maybe an ESP.

I have my Remote Controller working, but I want to be able to control it from HomeAssistant, so interfacing an ESP with it, I can use ESPHome.

I will also post the outcome here when I try.

Thank you again!!! \o/

**Comment by brunohenriquy** [comment]
@heit0r I guess my model is newer than yours, so it already comes with the connector and also the cable, it's intended to be plugged in to some compatible motherboards and be controlled inside Windows, so it was an awesome idea to use an external module to control it.

You were also right about this: "So, in practice, you only have d+ and the ground" 
The cable only uses two pins. It seems to be really easy to hook up a module and send signals to it.

![IMG_8700](https://github.com/user-attachments/assets/9edfbdc9-eaad-4b53-bb7f-b1243cd7238e)

![IMG_8701](https://github.com/user-attachments/assets/7a3139fc-269b-4019-8e77-a70070a02b92)

![IMG_8702](https://github.com/user-attachments/assets/714b4a71-3f22-41c4-9128-7f7c95ddd766)

![IMG_8703](https://github.com/user-attachments/assets/ff9ff555-ca71-4708-b854-42512422ba99)

**Comment by heit0r** [comment]
Thank you! You're right. This ARGB interface is made for communication with compatible motherboards, but also opens it for a lot of new possibilities. Controlling it with the ESP and it's built-in Wi-Fi capabilities makes this LED controller usable in a lot of other contexts than just the regular use on PC. Interesting to notice though, is that in your manual it says "indicator light" on what should be the IR led of the remote. So this gives us a clue that there's no IR being fired from the remote. Probably none of these LED controllers use IR remotes.

This is the on I got. You got the premium version :)
![th-2691010432](https://github.com/user-attachments/assets/77c75f59-ad4c-4de4-8d0c-c0bf80c7cb73)

**Comment by brunohenriquy** [comment]
Another approach would be to use an RF module like this one https://a.aliexpress.com/_mstX45i if the RF signal is fixed (which I think it is) and not rolling code.

Also, looking at this picture, we can conclude that RF really seems to be the standard for RGB strips:
![Screenshot 2024-08-28 at 15 55 24](https://github.com/user-attachments/assets/72f28d54-3d86-42b6-abba-341a1a431f3a)

**Comment by heit0r** [comment]
In one way or another, this module was a good buy because you were trying to clone the remote with IR but now you can still control the LED controller with RF. Interesting device, I'll take a better look at it.

**Comment by brunohenriquy** [comment]
> In one way or another, this module was a good buy because you were trying to clone the remote with IR but now you can still control the LED controller with RF. Interesting device, I'll take a better look at it.

Unfortunately mine is not this one, mine is just IR =/
I bought it a long time ago, I use it with some non-smart devices, so I tried with the Coolmoon controller.
Now I have two options:
- Buy this one IR/RF
- Develop one with ESP or Arduino

Anyway, this thread was a good talk! Thanks again! Will post what I ended up doing!
